gcr/add-staging-repo.sh

```diff
@@ -0,0 +1,107 @@
+#!/bin/sh
+#
+# This script is used to create a new "staging" repo in GCR.  Each sub-project
+# that needs to publish container images should have their own staging repo.
+#
+# Each staging repo exists in its own GCP project, and is writable by a
+# dedicated googlegroup.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+function usage() {
+    echo "usage: $0 <repo>" > /dev/stderr
+    echo "example:" > /dev/stderr
+    echo "  $0 coredns" > /dev/stderr
+    echo > /dev/stderr
+}
+
+function _color() {
+    tput setf $1
+}
+
+function _nocolor() {
+    tput sgr0
+}
+
+function color() {
+    _color $1
+    shift
+    echo "$@"
+    _nocolor
+}
+
+if [ $# != 1 ]; then
+    usage
+    exit 1
+fi
+if [ -z "$1" ]; then
+    usage
+    exit 2
+fi
+
+# The name of the sub-project being created, e.g. "coredns".
+REPO="$1"
+
+# The GCP project name.
+PROJECT="k8s-staging-${REPO}"
+
+# The GCS bucket that backs the GCR repo.
+BUCKET="artifacts.${PROJECT}.appspot.com"
+
+# The group that is admins all staging repos.
+ADMINS="k8s-infra-gcr-admins@googlegroups.com"
+
+# The group that can write to this staging repo.
+WRITERS="k8s-infra-gcr-staging-${REPO}@googlegroups.com"
+
+# The GCP org stuff needed to turn it all on.
+ORG="758905017065" # kubernetes.io
+BILLING="018801-93540E-22A20E"
+
+# Make the project, if needed
+if ! gcloud projects describe "${PROJECT}" >/dev/null 2>&1; then
+    color 6 "Creating project ${PROJECT}"
+    gcloud projects create "${PROJECT}" \
+        --organization "${ORG}"
+fi
+color 6 "Configuring billing for ${PROJECT}"
+gcloud beta billing projects link "${PROJECT}" \
+    --billing-account "${BILLING}"
+
+# Grant project viewer so the UI will work.
+color 6 "Granting project viewer to ${ADMINS}"
+gcloud \
+    projects add-iam-policy-binding "${PROJECT}" \
+    --member "group:${ADMINS}" \
+    --role roles/viewer
```

Comment:
do we just want admins as 'viewers'? what about 'owners'?

Reply:
They only need project viewer for the UI to work.  They have bucket admin for the specific buckets.

We're trying to keep untra-limited perms when possible.

Reply:
who is going to manage the projects? just who create them?

Reply:
the set of people who can create projects is very small and should, itself be governed by a googlegroup